### ISSUE TITLE
Bug: Backup fails for unknown reason in GS 2.16.2

### ISSUE THREAD

**Issue by bieganowski** [geosolutions-it/geoserver#186]
### Issue description

Geoserver's restore procedure sometimes ends in failure for no apparent reason. Log file attached for more info: [geoserver.log](https://github.com/geosolutions-it/geoserver/files/4438602/geoserver.log)
Issue was never present when the backup was created with no layers present in the Geoserver. With a layer loaded, issue occurred every time on the first backup execution after new environment creation, and sometimes afterwards.

### Environment

**Geoserver image**: geonode/geoserver:2.16.2
**Project**: [geosolutions-it/ADBPO-C041](https://github.com/geosolutions-it/ADBPO-C041) ([GeoNode/geonode-project](https://github.com/GeoNode/geonode-project) based project)
**GeoNode version**: 3

### Reproduction

1. Create an environment with `docker-compose`
2. Load a layer
3. Start backup with POST request `/rest/br/backup`:
```
{
   "backup":{
      "archiveFile":"geoserver_bk_file",
      "overwrite":"true",
      "options":{
         "option":[
            "BK_CLEANUP_TEMP=true"
         ]
      }
   }
}
```
4. Continue backup procedure with REST until Geoserver returns `FAILED` backup execution status

**Comment by afabiani** [comment]
I guess the title should be "**Backup** fails for unknown reason in GS 2.16.2".

**Comment by giohappy** [comment]
@afabiani and @bieganowski why "Backup" ? Isn't it an issue with restore?!!

**Comment by bieganowski** [comment]
No, the issue occurs when executing backup

**Comment by giohappy** [comment]
@afabiani clearly the fix implemented in https://github.com/GeoNode/geonode/pull/5974 is only a workaround. We need to investigate what is the root issue that makes 

1. first backup call fails sometimes 
2. if / why it only happens when Geoserver runs on Docker

**Comment by afabiani** [comment]
- JIRA [here](https://osgeo-org.atlassian.net/browse/GEOS-9567)
 - PR [here](https://github.com/geoserver/geoserver/pull/4169)

**Comment by giohappy** [comment]
just for curiosity, why did the backup succeed on second try @afabiani?

**Comment by afabiani** [comment]
If you have protected resources it fails every time, if you have public resources (in our case secured by GeoFenceAccessManager) the behavior depends basically by the state of the AccessManager and cache.

It is possible to easily reproduce the issue by removing the visibility to a resource to anonymous users.

Not sure which are steps and tests Michal has done. In my setup I never experencied this issue because I tested it always with all public resources.

**Comment by simboss** [comment]
can we close this one?

**Comment by giohappy** [comment]
I guess so... if I remember correctly the fixes were ported back, but in any case now we're targeting 2.17.x

**Comment by simboss** [comment]
@afabiani ?

**Comment by afabiani** [comment]
Yes, fixed with GS 2.17.2 currently active
